packages/flutter_tools/lib/src/flx.dart

```diff
@@ -34,14 +34,33 @@ const String _kFontSetMaterial = 'material';
 const String _kFontSetRoboto = 'roboto';
 
 class _Asset {
-  _Asset({ this.source, this.base, this.key });
+  _Asset({ this.base, String assetEntry, this.relativePath, this.source }) {
+    this._assetEntry = assetEntry;
+  }
+
+  String _assetEntry;
 
-  final String source;
   final String base;
-  final String key;
+
+  /// The entry to list in the generated asset manifest.
+  String get assetEntry => _assetEntry ?? relativePath;
+
+  /// Where the resource is on disk realtive to [base].
+  final String relativePath;
```

Comment:
Yup, from the one use of `source`, it looks like it's an alternative to `relativePath` (or what had been called `key`). +1 to a refactor and re-alignment